qiskit/synthesis/arithmetic/adders/rv_ripple_carry_adder.py

```diff
@@ -0,0 +1,156 @@
+         a_3: ──┼────┼────■──┤2       ├┤6          ├──┼────┼────■───────┤6 LAD_2 ├┤2          ├──┼────┼────┼────■──
+                │    │    │  │        ││           │  │    │    │       │        │└───────────┘┌─┴─┐  │    │    │
+         b_0: ──┼────┼────┼──┤        ├┤1 LAD_2_dg ├──┼────┼────┼───────┤1       ├─────────────┤ X ├──┼────┼────┼──
+              ┌─┴─┐  │    │  │  LAD_1 ││           │┌─┴─┐  │    │  ┌───┐│        │    ┌───┐    └───┘┌─┴─┐  │    │
+         b_1: ┤ X ├──┼────┼──┤        ├┤3          ├┤ X ├──┼────┼──┤ X ├┤3       ├────┤ X ├─────────┤ X ├──┼────┼──
+              └───┘┌─┴─┐  │  │        ││           │└───┘┌─┴─┐  │  ├───┤│        │    ├───┤         └───┘┌─┴─┐  │
+         b_2: ─────┤ X ├──┼──┤        ├┤5          ├─────┤ X ├──┼──┤ X ├┤5       ├────┤ X ├──────────────┤ X ├──┼──
+                   └───┘┌─┴─┐│        ││           │     └───┘┌─┴─┐└───┘└────────┘    └───┘              └───┘┌─┴─┐
+         b_3: ──────────┤ X ├┤        ├┤7          ├──────────┤ X ├───────────────────────────────────────────┤ X ├
+                        └───┘│        ││           │          └───┘                                           └───┘
+        cout: ───────────────┤3       ├┤8          ├───────────────────────────────────────────────────────────────
+                             └────────┘└───────────┘
+
+    Here *LAD_1* and *LAD_2* are the CX and CCX ladders respectively introduced in [1]. Note that
+    in this implementation the input register qubits are ordered as all qubits from
+    the first input register, followed by all qubits from the second input register.
+
+    Args:
+        num_qubits: The size of the register.
+
+    Returns:
+        The quantum circuit implementing the RV ripple carry adder.
+
+    Raises:
+        ValueError: If ``num_state_qubits`` is lower than 1.
+
+    **References:**
+
+    1. Remaud and Vandaele, Ancilla-free Quantum Adder with Sublinear Depth, 2025.
+    `arXiv:2501.16802 <https://arxiv.org/abs/2501.16802>`__
+
+    """
+
+    if num_qubits < 1:
+        raise ValueError("The number of qubits must be at least 1.")
+
+    qr_a = QuantumRegister(num_qubits, "a")
+    qr_b = QuantumRegister(num_qubits, "b")
+    qr_z = QuantumRegister(1, "cout")
+    qc = QuantumCircuit(qr_a, qr_b, qr_z)
+
+    if num_qubits == 1:
+        qc.ccx(qr_a[0], qr_b[0], qr_z[0])
+        qc.cx(qr_a[0], qr_b[0])
+        return qc
+
+    ab_interleaved = [q for pair in zip(qr_a, qr_b) for q in pair]
+
+    qc.cx(qr_a[1:], qr_b[1:])
+    qc.compose(_mcx_ladder(num_qubits - 1, 1), qr_a[1:] + qr_z[:], inplace=True)
+    qc.compose(_mcx_ladder(num_qubits, 2).inverse(), ab_interleaved + qr_z[:], inplace=True)
+    qc.cx(qr_a[1:], qr_b[1:])
+    qc.x(qr_b[1 : num_qubits - 1]) if num_qubits > 2 else qc.x(qr_b[1])
+    qc.compose(_mcx_ladder(num_qubits - 1, 2), ab_interleaved[:-1], inplace=True)
+    qc.x(qr_b[1 : num_qubits - 1]) if num_qubits > 2 else qc.x(qr_b[1])
```

Comment:
with this we can remove the pylint disable for `expression-not-assigned` 🙂 
```suggestion
    qc.x(qr_b[1 : max(2, num_qubits - 1)])
    qc.compose(_mcx_ladder(num_qubits - 1, 2), ab_interleaved[:-1], inplace=True)
    qc.x(qr_b[1 : max(2, num_qubits - 1)])
```

Reply:
Yes, you're right. Thanks!